src/directives/offense/poisonRoom.ts

```diff
@@ -0,0 +1,186 @@
+		if(this.pos.isVisible && this.room && !this.room.my && _.values(Overmind.colonies).length >= Game.gcl.level){
+			log.warning(`${this.print}: ${printRoomName(this.pos.roomName)} not enough GCL to contaminate room;`);
+			return;
+		}
+		if((this.room && this.room.hostiles.length > 0) || this.memory.isHostile){
+			this.memory.isHostile = true; //once hostile, always hostile, do not send scouts and always send defenders
+			this.overlords.defenders = new OutpostDefenseOverlord(this);
+		} else{
+			this.overlords.scout = new StationaryScoutOverlord(this);
+		}
+		this.overlords.roomPoisoner = new RoomPoisonerOverlord(this);
+	}
+
+	init() {
+		this.alert(`Poisoning Room ${this.pos.roomName}`);
+		//calculate wall positions
+		if(this.room && this.room.controller){
+			this.walkableSourcePosisions = _.filter(_.flatten(_.map(this.room.sources, s => s.pos.neighbors)),pos => pos.isWalkable(true));
+			this.walkableControllerPosisions =  _.filter(this.room.controller!.pos.neighbors, pos => pos.isWalkable(true));
+		}
+	}
+
+	private poison() {
+		//poison actions (creating wall csites are only done when roomPoisoner creep is present with > 0 energy)
+		let roomPoisoner = (!!this.overlords.roomPoisoner.roomPoisoners.length) ? this.overlords.roomPoisoner.roomPoisoners[0] : undefined;
+		if(roomPoisoner && roomPoisoner.carry.energy > 0){
+			//wall in sources
+			if(this.walkableSourcePosisions.length){
+				_.forEach(this.walkableSourcePosisions,pos=>{pos.createConstructionSite(STRUCTURE_WALL)});
+			}
+			//wall in controller, if option is selected
+			if(!this.memory.poisonSourcesOnly && this.walkableControllerPosisions.length){
+				_.forEach(this.walkableControllerPosisions,pos=>{pos.createConstructionSite(STRUCTURE_WALL)});
+			}
+		} else {
+			//if creep does not exist or .carry == 0, just remove the wall csite, they are not walkable and can block harvesting 
+			_.forEach(this.room!.constructionSites, csite => {csite.remove();} );
+		}
+	}
+	private isPoisoned(): boolean {
+		let result = false;
+		if (this.room && this.room.controller!.level > 1) {
+			result = !!this.walkableSourcePosisions && !this.walkableSourcePosisions.length;
+			if(!this.memory.poisonSourcesOnly){
+				result = result && !!this.walkableControllerPosisions && !this.walkableControllerPosisions.length;
+			}
+			return result;
+		} else {
+			return false;
+		}
+	}
+
+	run() {
+		if(Game.time % 25 == 0 && this.room && this.room.my){ //if visible and owned room
+			const roomRCL = this.room.controller!.level;
+			switch(roomRCL){
+				case 1:{
+					//remove any containers that can be next to sources
+					if(this.room.hostiles.length){
+						log.warning(`room ${this.print}: ${printRoomName(this.pos.roomName)} poisoning directive can't destory/remove structures/csites due to hostiles presense`);
+					} else {
+						if(this.room.containers.length){
+							_.forEach(this.room.containers, container => {container.destroy();});
```

Comment:
structure and csite operations should be on overlord, not directive. directives shouldn't do many state changing operations other than toggling overlords on and off

Reply:
go it, will do it an update you